File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isCorresponding(char a, char b){
        return (a == '{' && b == '}') || (a == '[' && b == ']') || (a == '(' && b == ')');
    }
    bool isValid(string s) {
        stack<char> theStack;
        for (char c: s){
            if (c == ']' || c == ')' || c == '}'){
                if (theStack.empty() || !isCorresponding(theStack.top(), c)){
                    return false;
                } else {
                    theStack.pop();
                }
            } else {
                theStack.push(c);
            }
        }
        return theStack.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
import java.util.Stack;
class Solution {
    public boolean isValid(String s) {
        Stack<Character> x = new Stack<Character>();
        for (int i=0; i<s.length(); ++i){
            if (s.charAt(i) == '(' || s.charAt(i) == '{' || s.charAt(i) == '['){
                x.push(s.charAt(i));
            } else {
                if (x.size() == 0){
                    return false;
                }
                char lastEl = x.peek();
                if ((lastEl == '(' && s.charAt(i) == ')') || (lastEl == '[' && s.charAt(i) == ']') || (lastEl == '{' && s.charAt(i) == '}')){
                    x.pop();
                } else {
                    return false;
                }
            }
        }
        return x.size() == 0;
        
    }
}